central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -249,6 +249,24 @@ func (s *serviceImpl) GetComplianceProfileResults(ctx context.Context, request *
 		return nil, errors.Wrapf(err, "Unable to retrieve compliance profile scan stats for %+v", request)
 	}
 
+	var ruleNames []string
+	for _, scanResult := range scanResults {
+		ruleNames = append(ruleNames, scanResult.RuleName)
+	}
+
+	controls, err := s.ruleDS.GetControlsByRuleName(ctx, ruleNames)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to retrieve compliance controls for rules %v", ruleNames)
+	}
+
+	for _, scanResult := range scanResults {
+		for _, control := range controls {
+			if scanResult.RuleId == control.RuleId {
+				scanResult.Controls = append(scanResult.Controls, control)
+			}
+		}
+	}
+
```

Comment:
This is the logic to enhance the checkresult data with Controls per checkresult.

**Different CO versions / benchmark versions on different clusters**
A challenge we face is different versions on different clusters.

A checkresult is aggregated and could implement multiple controls.
These controls are versioned by benchmark (e.g. CIS1.4.0, CIS1.5.0).
Currently we do not distinguish for the Benchmark version.


**Which controls to display?**
Derive Benchmark from Profile and filter controls by the Benchmark implemented in the Profile.
We can do the same to filter for version. This requires us to pin all profiles to a benchmark version, we'll not implement the latest versions.


**Conclusion**
I am missing the whole picture how much impact it has if we do not distinguish control data by version. 
In my opinion it doesn't feel like extremely important and rarely  occur.